ide/lsp.client/src/org/netbeans/modules/lsp/client/bindings/CompletionProviderImpl.java

```diff
@@ -226,36 +223,66 @@ public void render(Graphics grphcs, Font font, Color color, Color color1, int i,
-                                return new CompletionTask() {
+                                return new AsyncCompletionTask(new AsyncCompletionQuery() {
                                     @Override
-                                    public void query(CompletionResultSet resultSet) {
-                                        resultSet.setDocumentation(new CompletionDocumentation() {
-                                            @Override
-                                            public String getText() {
-                                                return documentation;
-                                            }
-                                            @Override
-                                            public URL getURL() {
-                                                return null;
-                                            }
-                                            @Override
-                                            public CompletionDocumentation resolveLink(String link) {
-                                                return null;
+                                    protected void query(CompletionResultSet resultSet, Document doc, int caretOffset) {
+                                        CompletionItem resolved;
+                                        if ((i.getDetail() == null || i.getDocumentation() == null) && hasCompletionResolve(server)) {
+                                            CompletionItem temp;
+                                            try {
+                                                temp = server.getTextDocumentService().resolveCompletionItem(i).get();
+                                            } catch (InterruptedException | ExecutionException ex) {
+                                                Exceptions.printStackTrace(ex);
+                                                temp = i;
                                             }
-                                            @Override
-                                            public Action getGotoSourceAction() {
-                                                return null;
-                                            }
-                                        });
+                                            resolved = temp;
+                                        } else {
+                                            resolved = i;
+                                        }
+                                        if (resolved.getDocumentation() != null || resolved.getDetail() != null) {
+                                            resultSet.setDocumentation(new CompletionDocumentation() {
+                                                @Override
+                                                public String getText() {
+                                                    StringBuilder documentation = new StringBuilder();
+                                                    documentation.append("<html>\n");
```

Comment:
should add `NOI18N`s?

Reply:
Probably yes, I'll try to remember that and add that on next update.